x-pack/plugins/observability_ai_assistant/server/functions/recall.ts

```diff
@@ -87,12 +87,17 @@ export function registerRecallFunction({
         messages.filter((message) => message.message.role === MessageRole.User)
```

Comment:
nit: this is a good use case for `findLast` (well supported and not nearly used enough :D )

```suggestion
        messages.findLast((message) => message.message.role === MessageRole.User)
```

Reply:
I did not know this exists 😄 unfortunately TS doesn't think it exists (??)